Common/Data/Consolidators/PeriodCountConsolidatorBase.cs

```diff
@@ -214,13 +214,14 @@ public override void Update(T data)
                     }
                 }
 
-                OnDataConsolidated(_workingBar);
+                // Set _lastEmit first because OnDataConsolidated will set _workingBar to null
                 _lastEmit = IsTimeBased && _workingBar != null ? _workingBar.Time.Add(Period ?? TimeSpan.Zero) : data.Time;
```

Comment:
Here `_lastEmit` should be rounded? But bellow in `public override void Scan(DateTime currentLocalTime)` we set `_lastEmit` with a different logic, just `currentLocalTime` which **might not be rounded the bug right**? if it also used the bar time the issue shouldn't happen?